lib/cm/client.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Created by kugajjar on 6/3/15.
+ */
+
+'use strict';
+
+var crypto          = require('crypto'),
+    request         = require('request'),
+    HttpStatus      = require('http-status-codes'),
+    url             = require('url'),
+    Q               = require('q');
+
+module.exports = function client(nemo) {
+
+    var config = nemo._config.get('cm');
+
+    var generateToken = function (timeStamp) {
```

Comment:
This doesn't really matter, but you could define this function in the scope of the `authentication_params` function because that's the only place it's used.  That'd keep it only in the scope where it's used. It makes no difference -- just stylistic.